src/System.Transactions.Local/src/System/Transactions/TransactionState.cs

```diff
@@ -1533,7 +1516,7 @@ internal override void CreateAbortingClone(InternalTransaction tx)
 
         internal override void GetObjectData(InternalTransaction tx, SerializationInfo serializationInfo, StreamingContext context)
         {
-            throw CreateTransactionAbortedException(tx);
+            throw new PlatformNotSupportedException();
```

Comment:
I missed that the methods in this file aren't ISerializable implementations. I've put their implementations back.